src/browser/cdp/selectivity/file-hash-writer.ts

```diff
@@ -0,0 +1,165 @@
+import { memoize } from "lodash";
+import path from "node:path";
+import fs from "fs-extra";
+import { FileHashProvider } from "./file-hash-provider";
+import { shallowSortObject } from "./utils";
+import type { NormalizedDependencies } from "./types";
+
+export class FileHashWriter {
```

Comment:
tracks up-to-date hashes for all files:
<img width="643" height="575" alt="image" src="https://github.com/user-attachments/assets/6f8c2f22-f5e9-46a5-b578-d441a1754106" />